addon/mixins/log-api-errors.js

```diff
@@ -0,0 +1,41 @@
+import Ember from 'ember';
+
+const {
+    get,
+    canInvoke,
+    inject: { service }
+} = Ember;
+
+export default Ember.Mixin.create({
+    errorManager: service('error-manager'),
+
+    handleResponse(status, headers, payload, requestData) {
+        let result = this._super(...arguments);
+
+        if(this.isSuccess(status, headers, payload)) {
+            return result;
+        }
+
+        //provide a more detailed message (request/response info, ect) for ALL error types
+        result.message = this.formatErrorMessage(result, status, headers, payload, requestData);
+        get(this, 'errorManager').consume(result);
+
+        return result;
+    },
+
+    formatErrorMessage(error, status, headers, payload, requestData) {
+        if(typeof payload !== 'string') {
+            payload = JSON.stringify(payload);
+        }
+
+        if(canInvoke(this, 'generatedDetailedMessage')) {
```

Comment:
@bakerac4 I thought about that but I think in this case I need to specifically test if the method exists, because otherwise the error message needs to be returned as-is, in that final `else` block. Unless I check to see if the value returned by each `tryInvoke` is `undefined` or not, e.g.

```javascript
let formattedMsg;

formattedMsg = tryInvoke(this, 'generatedDetailedMessage', [...]);
if(formattedMsg) {
  return formattedMsg
}

formattedMsg = tryInvoke(this, 'generateDetailedMessage', [...]);
if(formattedMsg) {
  return formattedMsg
}

return error.message;
```